addons/website_slides/models/slide_channel.py

```diff
@@ -773,9 +787,22 @@ def _remove_membership(self, partner_ids):
         users = self.env['res.users'].sudo().search([
             ('partner_id', 'in', list(earned_karma)),
         ])
+        karma_values = {}
         for user in users:
-            if earned_karma[user.partner_id.id]:
-                user.add_karma(-1 * earned_karma[user.partner_id.id])
+            # Group the earned karma per channel
+            earned_karma_per_channel = defaultdict(lambda: 0)
+            for entry in earned_karma[user.partner_id.id]:
+                if entry['karma']:
+                    earned_karma_per_channel[entry['channel_id']] += entry['karma']
+            # Removed the earned karma for each channel
+            for channel, karma in earned_karma_per_channel.items():
```

Comment:
I don't think this will work as it will override the same "user" key multiple times.

Probably needs a specific test to make sure.